hugo/content/guides/keywords-as-identifiers/_index.md

```diff
@@ -0,0 +1,164 @@
+```
+
+Roughly summarized, the background for this behaviour is, that Langium internally use [Chevrotain](https://chevrotain.io) for tokenizing, i.e. splitting text into single tokens, e.g. words separated by white space.
+Chevrotain uses regex for splitting text into tokens.
+Since keywords are realized as regex as well, *all* occurrances of "Hello" are treated as keywords for the parser rule for greetings,
+even "Hello" which are intented to be names, which causes the parsing problems (see the two errors in the "Problems" tab above).
+
+In order to explicitly enable parsing "Hello" as name as well, tweak the parser rule for persons in this way:
+
+```langium
+Person: 'person' name=(ID | 'Hello');
+terminal ID: /[_a-zA-Z][\w_]*/; // the terminal rule for ID is unchanged!
+```
+
+Now Langium knows, that "Hello" not always indicates the keyword of the greeting parser rule, but can also occur as explicit value for the `name` property of the parser rule for persons.
+That's it! (Don't forget to run `npm run langium:generate` after updating the grammar.)
+
+![screenshot with fixed grammar](fixed-1-grammar.png)
+
+Since the `name` property is used for cross-references by the parser rule for greetings, "Hello" needs to be supported here as well. For that, we recommend to introduce a [data type rule](/docs/grammar-language/#data-type-rules) like "PersonID" in the example, since it makes it easier to support more keywords in the future:
+
+```langium
+Person: 'person' name=PersonID;
+
+Greeting: 'Hello' person=[Person:PersonID] '!';
+
+PersonID returns string: ID | 'Hello';
+```
+
+
+As you can see, Langium accepts "Hello" as value for person's names now.
+Nevertheless, the name "Hello" still is highlighted in blue and looks like a keyword "Hello". This leads us to the second step.
+
+
+## Step 2: Change the semantic type of the resulting token
+
+The __second step__ is to change the semantic type of the resulting token in order to adjust the highlighting in the editor:
+While parsing text with Langium is done in a language server, the highlighting is done in editors (the language clients).
+Editors like VS Code usually use *syntax highlighting* basing on the tokenized text. This highlighting can be complemented by *semantic highlighting* with additional semantic information for the tokens from the language server.
+
+In case of Langium and VS Code, VS Code uses by default TextMate grammars, which can be seen as collections of regex (and which is generated by `npm run langium:generate`), to split the text into tokens and assigns a (syntactic) type to these tokens. The color for highlighting the token is chosen depending on the assigned type.
+In the example, a regex for the "Hello" keyword matches *all* strings "Hello" in text, resulting in the blue color even for "Hello" used as name.
+
+In contrast to this, Langium is able to distinguish "Hello" keywords and "Hello" names and therefore is able to assign different *semantic* types to "Hello" tokens.
```

Comment:
That's not correct AFAIK, see above. It just accepts the keyword's token text as a value.

Reply:
Sorry, I don't see the problem yet: In my eyes, the AST is created on top of the token stream by applying the parser rules. By looking into the AST, you can distinguish, for which purpose tokens are used.

I tried to formulate the sentence more clear.

Reply:
Right, you also right to some extend. Your new formulation is very good and unambiguous.

Reply:
Thank you!